Source/Core/Request.swift

```diff
@@ -167,13 +171,24 @@ public struct Request {
     
     /**
      Intialize a request value.
+
+     - parameter method: The HTTP request method.
+     - parameter url: The URL string of the HTTP request.
+     */
+    init(_ method: Method, url: URL) {
+        self.method = method
+        self.requestURL = url
+    }
+
+    /**
+     Intialize a request value.
      
      - parameter method: The HTTP request method.
      - parameter url: The URL string of the HTTP request.
     */
-    init(_ method: Method, url: String) {
-        self.method = method
-        self.url = url
+    init(_ method: Method, url urlString: String) {
```

Comment:
If we ditch `constructURLString` in favor of `constructURL` we could just remove this initializer since it isn't a public API anyway.